src/suggestion-box.ts

```diff
@@ -0,0 +1,482 @@
+        const template = this.template_slots[0];
+        if (template) {
+            this._templateContent = template.innerHTML;
+        }
+    }
+
+    private _indexOfTarget(event: any) {
+        let el = event.target;
+        while (el) {
+            if (el.slot == "__private-item") break;
+            el = el.parentElement;
+        }
+        return this.private_items.indexOf(el);
+    }
+
+    private _mouseOverEvent(e: any) {
+        if (e.screenX == this.screenx && e.screenY == this.screeny) return;
+        let curr_index = this._indexOfTarget(e);
+        this.screenx = e.screenX;
+        this.screeny = e.screenY;
+        this.index = curr_index;
+        if (curr_index == -1) return;
+        if (curr_index >= 0) {
+            this.private_items[this.index].classList.add(`${this.activecls}`);
+        }
+        if (
+            this.private_items.length > 0 &&
+            curr_index !== this.private_items.length - 1
+        ) {
+            this.private_items[this.private_items.length - 1].classList.remove(
+                `${this.activecls}`
+            );
+        }
+
+        this.prevIndex = curr_index;
+        this._scrollIntoView();
+    }
+
+    private _scrollIntoView() {
+        const selectedItem = this.private_items[this.index];
+        if (selectedItem) {
+            selectedItem.scrollIntoView({
+                behavior: "instant",
+                block: "nearest",
+                inline: "nearest",
+            });
+        }
+    }
+
+    private _inputSelectEvent() {
+        this.index = -1;
+    }
+
+    private _inputFocusEvent() {
+        this.show_container = true;
+        if (
+            this.filtered_list.length > 0 &&
+            (this.input_el as HTMLInputElement)?.value.length >= this.minchar
+        ) {
+            this.show_container = true;
+            this.show_match = true;
+            let el = document.createElement("div");
+            el.className = "sgn-match";
+            el.slot = "sgn-match";
+            el.innerHTML = "No matching result";
+            this.appendChild(el);
+            this.setSuggestion(this.filtered_list, this.has_more);
+        }
+        this.index = -1;
+    }
+
+    private _mouseOutEvent(e: any) {
+        if (e.screenX == this.screenx && e.screenY == this.screeny) return;
+        if (this.prevIndex >= 0 && this.prevIndex < this.result.length) {
+            this.private_items[this.index]?.classList.remove(
+                `${this.activecls}`
+            );
+        }
+    }
+    windowClickEvent(event: any) {
+        let el = event.target;
+        while (el) {
+            if (el == this) return;
+            el = el.parentElement;
+        }
+        this.show_container = false;
+    }
+    private _format(obj: any) {
+        if (!this._templateContent) return "";
+        let str = this._templateContent;
+        str = str.replace(/\{([^}]+)\}/g, (match, key) => {
+            if (obj.hasOwnProperty(key.trim())) {
```

Comment:
Save key.trim() in a local variable and make use of that variable instead of calling key.trim() always.